test-gateways/src/test/groovy/com/mx/gateway/GatewayMethodSpec.groovy

```diff
@@ -0,0 +1,55 @@
+package com.mx.gateway
+
+
+import com.mx.testing.AccountAccessorImpl
+import com.mx.testing.BaseAccessorImpl
+import com.mx.testing.gateway.api.AccountGateway
+import com.mx.testing.model.Account
+
+import spock.lang.Specification
+
+/**
+ * These tests are intended to check that the internals of the generated gateway methods work correctly and don't
+ * encounter any casting issues.
+ */
+class GatewayMethodSpec extends Specification {
+  AccountGateway subject
+
+  def setup() {
+    def baseAccessor = new BaseAccessorImpl(null)
+    baseAccessor.setAccounts(new AccountAccessorImpl(null))
+    subject = AccountGateway.builder().baseAccessor(baseAccessor).build()
+  }
+
+  def "invocation of list type"() {
+    when:
+    def result = subject.get20240101("123")
+
+    then:
+    result != null
+    result.getResult() instanceof com.mx.testing.models.v20240101.Account
```

Comment:
You can directly access properties in groovy ie `result.result` instead of `result.getResult()`